### ISSUE TITLE
searching for nodes after Shift-A -> error

### ISSUE THREAD

**Issue by luckychris** [nortikin/sverchok#4294]
On Macbook, Blender 3.0 in Sverchok version 1.0 with latest update i cannot search anymore for nodes, got this error:

![image](https://user-images.githubusercontent.com/232549/129960973-08cf11a3-4645-4cfe-bf7b-b6052dafb162.png)

**Comment by luckychris** [comment]
but this works in 2.92

**Comment by zeffii** [comment]
i can't reproduce that

**Comment by luckychris** [comment]
did you try on macos?

**Comment by Durman** [comment]
I also have a problem with searching nodes about 3 last days, was waiting whether will anybody report about the problem too. The problem does not appear when Sverchok is just lunched but after adding some nodes after a while.

**Comment by zeffii** [comment]
my guess is that there are one or more nodes without a docstring, which happen to get triggered by certain search input

**Comment by zeffii** [comment]
just to be clear, this is happening on both 3.0 and below?

**Comment by luckychris** [comment]
> just to be clear, this is happening only in 3.0?

yes

**Comment by zeffii** [comment]
if you ever get that again, try to take a screenshot of the characters you entered into the Search field.

**Comment by zeffii** [comment]
otherwise i will commit something horrible to print out more information if that error happens again..
```python
def ensure_valid_show_string(nodetype):

    try:
        loop_reverse[nodetype.bl_label] = nodetype.bl_idname
        description = nodetype.bl_rna.docstring.get_shorthand()
        return nodetype.bl_label + ensure_short_description(description)
    except Exception as err:
        print(f"{nodetype}")
        print(err)   
```

**Comment by zeffii** [comment]
> note to self.

not all nodes have `bl_rna`..  like dummy representations of optional nodes..
```python
import bpy
import sverchok
from sverchok.menu import make_node_cats

cats = make_node_cats()

for cat, elements in cats.items():
    print(cat)
    for element in elements:
        name = element[0]
        if not name in {'separator', 'NodeReroute'}:
            print('..', name)
            node_ref = bpy.types.Node.bl_rna_get_subclass(name)
            try:
                doc = node_ref.bl_rna.docstring
            except Exception as err:
                print(name)
                print(err)
        
```
gives... a snippet
```console

Exchange
.. SvExNurbsToJsonNode
SvExNurbsToJsonNode
'NoneType' object has no attribute 'bl_rna'
.. SvExJsonToNurbsNode
SvExJsonToNurbsNode
'NoneType' object has no attribute 'bl_rna'
.. SvImportSolidNode
SvImportSolidNode
'NoneType' object has no attribute 'bl_rna'
.. SvExportSolidNode
SvExportSolidNode
'NoneType' object has no attribute 'bl_rna'
.. SvReceiveFromSorcarNode
```
if it has no `bl_rna` it's not going to have `bl_rna.docstring`

**Comment by zeffii** [comment]
kind of hard to debug if I can't reproduce the exact error.

**Comment by Durman** [comment]
I have this error but I can't reproduce it.
```
2021-09-08 15:57:42,647 [ERROR] sverchok.utils.sv_extra_search 68: Nodetype "<class 'sverchok.nodes.solid.bound_box.SvSolidBoundBoxNode'>": ensure_valid_show_string() threw an exception:
 'Node' object has no attribute 'docstring'
TypeError: EnumProperty(...): expected a tuple containing (identifier, name, description) and optionally an icon name and unique number
File "...\sverchok\utils\sv_extra_search.py", line 137, in item_cb
```

**Comment by zeffii** [comment]
sv_extra_search was not extensively tested in the presence of dummynodes. maybe i have time tomorrow to investigate.
i'll Filter nodes with attribute `missing_dependency`.

**Comment by zeffii** [comment]
@Durman  that's bizarre because if you run this in the console.. you'll see the same as me?
```console
import sverchok
n = sverchok.nodes.solid.bound_box.SvSolidBoundBoxNode
n.bl_rna.docstring
```
outputs
```console
<sverchok.utils.docstring.SvDocstring object at 0x000002CB54683FA0>
```

still a puzzle though.

**Comment by zeffii** [comment]
the extra search operator only calls `gather_items` once when the operatos is invoked. And the result is stored in 

```console
sverchok.utils.sv_extra_search.loop[" results"]
```
and that's where `item_cb` gets the data, i'm surprised this is happening. Alternatively it is possible to add a placeholder text for docstring instead of skipping. We might learn more.

instead of the new
```python
            docstring = ensure_valid_show_string(nodetype)
            if not docstring:
                continue

            fx.append((str(idx), docstring, '', idx))
```
it would be 
```python
            docstring = ensure_valid_show_string(nodetype)
            if not docstring:
                docstring = "internal error"    # or nodetype.name

            fx.append((str(idx), docstring, '', idx))
```
but that still wont explain _why_ you get that error.

**Comment by zeffii** [comment]
ok.. i don't know if this actually resolves this issue #4294 ,  but realistically it has a good chance.  If problem persists, reopen.
